Fix #13222 : Vehicle Collision causes negative number of passengers

raacker opened the pull request:
when peep_sprite_remove is called, RemoveFromRide() will decrease ride's num_riders

duncanspumpkin commented:
I think this will need a changelog entry. I'm pretty sure it's an original bug from my look at the assembly.

duncanspumpkin commented:
Also will need network increment.

raacker commented:
@duncanspumpkin Could you tell me what do you mean by "network increment"?

duncanspumpkin commented:
Network.cpp has a version number just increment the number by one. It prevents folk from getting desyncs due to different simulation states as they won't be able to connect with different network versions

tupaschoal commented:
I think it is `NetworkBase.cpp` now, but I'm not 100% sure

raacker commented:
@duncanspumpkin I see thanks. Just updated network version number.
@tupaschoal And it was correct. NetworkBase.cpp has NETWORK_STREAM_VERSION

@Broxzier Hey I also added this fix to changelog because it has been there for a while even before 3.1.0. Is it correct then?

Broxzier commented:
> @Broxzier Hey I also added this fix to changelog because it has been there for a while even before 3.1.0. Is it correct then?

Yep!

AaronVanGeffen commented on distribution/changelog.txt:
```diff
@@ -33,6 +33,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222]  Vehicle Collision causes negative number of passengers.
```
Stray space:
```suggestion
- Fix: [#13222] Vehicle Collision causes negative number of passengers.
```

raacker replied:
Thanks for the good catch :)

Gymnasiast commented on distribution/changelog.txt:
```diff
@@ -34,6 +34,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222] Vehicle Collision causes negative number of passengers.
```
```suggestion
- Fix: [#13222] Vehicle collision causes negative number of passengers.
```

Gymnasiast commented on distribution/changelog.txt:
```diff
@@ -34,6 +34,7 @@
 - Fix: [#13129] Missing error message when waiting for train to leave station on the ride measurements graph.
 - Fix: [#13138] Fix logical sorting of list windows.
 - Fix: [#13158] Cursors are drawn incorrectly in text input fields.
+- Fix: [#13222] Vehicle collision causes negative number of passengers.
```
Aw, shite, should have added ` (original bug)`. :/
